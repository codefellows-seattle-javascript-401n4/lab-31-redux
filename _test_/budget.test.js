'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import Jest from 'jest';
import uuid from 'uuid/v1';
import CatForm from '../src/components/catForm.js';



test('create a new budget with the name groceries and the budget amount to be ', () => {
  let name = 'groceries';
  let isEditing = false;
  let budget = 20;
  let createBudget = (state) => {
    expect(CatForm.state().name).toBe('groceries');
    expect(CatForm.state().isEditing).toBe(false);
    expect(CatForm.state().budget).toBe(20);
  };
  const component = renderer.create(
    <CatForm />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
}
);

test('test should fail because of change in budget content', () => {
  const component = renderer.create(
    <CatForm />
  );

  const other = renderer.create(
    <CatForm name="games"/>
  );

  let tree = component.toJSON();
  let otherTree = other.toJSON();
  expect(tree).toMatchSnapshot();
  expect(otherTree).toEqual(tree);
});

test('should pass with unique ids',() => {
  const component = renderer.create(
    <CatForm id='2222'/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should fail with non unique ids', () => {
  const component = renderer.create(
    <CatForm id='2222'/>
  );
  const other = renderer.create(
    <CatForm id='2222'/>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(other).toBe(tree);

});
