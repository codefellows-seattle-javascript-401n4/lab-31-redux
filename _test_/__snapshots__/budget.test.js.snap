// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create a new budget with the name groceries and the budget amount to be  1`] = `
<div>
  <form
    className="form"
    onSubmit={[Function]}
  >
    Your budget name:
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    Your budget expense:
    <input
      onChange={[Function]}
      type="text"
      value={0}
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`should fail with non unique ids 1`] = `
<div>
  <form
    className="form"
    onSubmit={[Function]}
  >
    Your budget name:
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    Your budget expense:
    <input
      onChange={[Function]}
      type="text"
      value={0}
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`should pass with unique ids 1`] = `
<div>
  <form
    className="form"
    onSubmit={[Function]}
  >
    Your budget name:
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    Your budget expense:
    <input
      onChange={[Function]}
      type="text"
      value={0}
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`test should fail because of change in budget content 1`] = `
<div>
  <form
    className="form"
    onSubmit={[Function]}
  >
    Your budget name:
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    Your budget expense:
    <input
      onChange={[Function]}
      type="text"
      value={0}
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
